src/app/(app)/missing-shoes/Components/MissingShoesFormBuilder.tsx

```diff
@@ -0,0 +1,261 @@
+});
+
+export type MissingSchemaType = z.infer<typeof missingShoesSchema>;
+
+export const MissingShoesFormBuilder = () => {
+  const phoneNumberMask = useMask({
+    mask: "___________",
+    replacement: { _: /\d/ },
+  });
+  const cnicMask = useMask({
+    mask: "3____-_______-_",
+    replacement: { _: /\d/ },
+  });
+
+  const form = useForm<MissingSchemaType>({
+    resolver: zodResolver(missingShoesSchema),
+    defaultValues: {
+      color: "",
+      type: ShoesTyes.Formal,
+      size: "",
+      time: undefined,
+      ownerName: "",
+      ownerPhoneNumber: "",
+      ownerCnic: "",
+    },
+    mode: "all",
+  });
+
+  const SHOES_TYPES = [
+    ShoesTyes.Boots,
+    ShoesTyes.DressShoes,
+    ShoesTyes.Sandals,
+    ShoesTyes.Sneakers,
+    ShoesTyes.Formal,
+    ShoesTyes.Other,
+  ];
+
+  const {
+    formState: { errors, isValid },
+  } = form;
+
+  function onSubmit(values: z.infer<typeof missingShoesSchema>) {
+    return values
+  }
+  return (
+    <FormWrapper>
+      <FormTitle
+        title="Report Missing Shoe"
+        description="Please provide details about the missing shoe"
+      />
+      <Form {...form}>
+        <form
+          data-testid="leaveRequestForm"
+          action={() => form.handleSubmit(onSubmit)()}
+          className="space-y-4"
+        >
+          <div className="grid grid-cols-1 lg:grid-cols-2 gap-4">
+            <FormField
+              control={form.control}
+              name={"color"}
+              render={({ field }) => (
+                <FormItem>
+                  <FormLabel>Color</FormLabel>
+                  <FormControl>
+                    <Input placeholder="e.g., Red, Blue, Black" {...field} />
+                  </FormControl>
+                  <FormMessage />
+                </FormItem>
+              )}
+            />
+            <FormField
+              control={form.control}
+              name="type"
+              render={({ field }) => (
+                <FormItem>
+                  <FormLabel>Shoes Type</FormLabel>
+                  <FormControl>
+                    <Select
+                      {...field}
+                      value={field.value}
+                      onValueChange={field.onChange}
+                    >
+                      <SelectTrigger data-testid="leave-type" id="type">
+                        <SelectValue placeholder="Select shoe type" />
+                      </SelectTrigger>
+                      <SelectContent>
+                        {SHOES_TYPES.map((type) => (
+                          <SelectItem key={type} value={type}>
+                            {type}
+                          </SelectItem>
+                        ))}
+                      </SelectContent>
+                    </Select>
+                  </FormControl>
+                  <FormMessage />
+                </FormItem>
+              )}
+            />
+          </div>
+          <div className="grid grid-cols-1 lg:grid-cols-2 gap-4">
```

Comment:
can we remove grid and use only one input in one row

Reply:
Fixed